Cannot use `gpg --import private_key.gpg` directly in GitHub Actions as directed by the doc

<!-- Thanks for reporting an issue! Please make sure you click the link above to view the issue guidelines, then fill out the blanks below. -->

What are the steps to reproduce this issue?
-------------------------------------------
1. Follow the documentation to create a private key and put it in GitHub's secrets with the name `GPG_PRIVATE_KEY`
2. Use this minimum reproducible yaml to set up GitHub Actions

```yaml
name: demo
on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]
jobs:
  demo_error_on_gpg_import:
    runs-on: ubuntu-latest
    env:
      GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}$
    steps:
      - name: Setup git secret
        run: |
          sudo apt-get update && sudo apt-get install -y git-secret
          echo $GPG_PRIVATE_KEY | tr ',' '\n' > ./private_key.gpg
          gpg --import private_key.gpg
```
3. Trigger GitHub Actions

What happens?
-------------
It seems like `gpg --import` requires command line interaction, which GitHub Actions cannot provide. Hence, the `Inappropriate ioctl for device` error is thrown.

What were you expecting to happen?
----------------------------------
`gpg --import private_key.gpg` can be executed on GitHub Actions without problem. 

Any logs, error output, etc?
----------------------------
```
gpg: key 9E22ADA9E6532937: public key "gcp_related GitHub <gcp_related@github.com>" imported
gpg: key 9E22ADA9E6532937/9E22ADA9E6532937: error sending to agent: Inappropriate ioctl for device
gpg: key 9E22ADA9E6532937/59E87A3B8BF021B7: error sending to agent: Inappropriate ioctl for device
gpg: error reading 'private_key.gpg': Inappropriate ioctl for device
gpg: import from 'private_key.gpg' failed: Inappropriate ioctl for device
gpg: Total number processed: 0
gpg:               imported: 1
gpg:       secret keys read: 1
Error: Process completed with exit code 2.
```

Any other comments?
-------------------
This problem can be resolved by explicitly telling `gpg` not to allow command line interaction, i.e.

```
gpg --batch --yes --pinentry-mode loopback --import private_key.gpg
```

If we replace `gpg --import private_key.gpg` with the line above, no error is thrown. Therefore, I hope the documentation can be updated to reflect this issue.

What versions of software are you using?
----------------------------------------
**Operating system:** (`uname -a`) 
The version reported is the one on the virtual machine of GitHub Actions.

Linux fv-az193-897 5.4.0-1047-azure #49-Ubuntu SMP Thu Apr 22 14:30:37 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

**`git-secret` path:** (`which git-secret`) 
/usr/bin/git-secret

**`git-secret` version:** (`git secret --version`) 
0.2.2

**`git` version:** (`git --version`) 
git version 2.31.1

**Shell type and version:** (`$SHELL --version`) 
GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)

**`gpg` version:** (`gpg --version`) 
gpg (GnuPG) 2.2.19

<!-- Love git-secret? Please consider supporting our collective:
👉  https://opencollective.com/git-secret/donate -->

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -18,7 +18,7 @@
 
 - Rename `killperson` command to `removeperson` (#684)
 - Moves `file_has_line` utility to tests and fixes how it is used
-- Refactors docs: new pages, new content
+- Refactors docs: new pages, new content (#675)
 
 
 ## 0.4.0

--- a/man/man7/git-secret.7.md
+++ b/man/man7/git-secret.7.md
@@ -90,12 +90,12 @@ if your app is called MyApp and your CI/CD provider is CodeShip. It is easier no
 # see: https://git-secret.io/installation
 
 # Create private key file
-echo $GPG_PRIVATE_KEY > ./private_key.gpg
+echo "$GPG_PRIVATE_KEY" > ./private_key.gpg
 # Import private key and avoid the "Inappropriate ioctl for device" error
 gpg --batch --yes --pinentry-mode loopback --import private_key.gpg
 # Reveal secrets without user interaction and with passphrase. If no passphrase
 # is created for the key, remove `-p $GPG_PASSPHRASE`
-git secret reveal -p $GPG_PASSPHRASE
+git secret reveal -p "$GPG_PASSPHRASE"
 # carry on with your build script, secret files are available ...
 ```
 
@@ -108,7 +108,7 @@ gpg --armor --export-secret-key myapp@codeship.com | tr '\n' ','
 You can then create your private key file with:
 
 ```shell
-echo $GPG_PRIVATE_KEY | tr ',' '\n' > ./private_key.gpg
+echo "$GPG_PRIVATE_KEY" | tr ',' '\n' > ./private_key.gpg
 ```
 
 Also note: the `gpg` version on the CI/CD server **MUST MATCH** the one used locally. Otherwise, `gpg` decryption can fail silently, which leads to `git secret reveal` reporting `cannot find decrypted version of file` error. To be specific, `apt-get install gnupg` points to version [2.2.20](https://packages.ubuntu.com/impish/gnupg), yet `brew install gnupg` points to version [2.3.4](https://formulae.brew.sh/formula/gnupg) (as of 2022-01-17). Thus a `git-secret` encrypted file on macOS using the latest `gpg` installed from `brew` cannot be decrypted on Ubuntu (e.g. GitHub Actions' latest Ubuntu machine) using the latest `gpg` installed from `apt-get`. The work-around for this specific case is to downgrade `gpg` with `brew install gnupg@2.2.33`.

--- a/man/man7/git-secret.7.md
+++ b/man/man7/git-secret.7.md
@@ -80,9 +80,9 @@ with the changes in your code.
 One way of doing it is the following:
 
 1. [create a gpg key](#using-gpg) for your CI/CD environment. You can chose any name and email address you want: for instance `MyApp CodeShip <myapp@codeship.com>`
-if your app is called MyApp and your CI/CD provider is CodeShip. It is easier not to define a password for that key.
+if your app is called MyApp and your CI/CD provider is CodeShip. It is easier not to define a passphrase for that key. However, if defining a passphrase is unavoidable, use a unique passphrase for the private key.
 2. run `gpg --armor --export-secret-key myapp@codeship.com` to get your private key value
-3. Create an env var on your CI/CD server `GPG_PRIVATE_KEY` and assign it the private key value.
+3. Create an env var on your CI/CD server `GPG_PRIVATE_KEY` and assign it the private key value. If a passphrase has been setup for the private key, create another env var on the CI/CD server `GPG_PASSPHRASE` and assign it the passphrase of the private key.
 4. Then write your Continuous Deployment build script. For instance:
 
 ```shell
@@ -91,10 +91,11 @@ if your app is called MyApp and your CI/CD provider is CodeShip. It is easier no
 
 # Create private key file
 echo $GPG_PRIVATE_KEY > ./private_key.gpg
-# Import private key
-gpg --import ./private_key.gpg
-# Reveal secrets
-git secret reveal
+# Import private key and avoid the "Inappropriate ioctl for device" error
+gpg --batch --yes --pinentry-mode loopback --import private_key.gpg
+# Reveal secrets without user interaction and with passphrase. If no passphrase
+# is created for the key, remove `-p $GPG_PASSPHRASE`
+git secret reveal -p $GPG_PASSPHRASE
 # carry on with your build script, secret files are available ...
 ```
 
@@ -109,6 +110,9 @@ You can then create your private key file with:
 ```shell
 echo $GPG_PRIVATE_KEY | tr ',' '\n' > ./private_key.gpg
 ```
+
+Also note: the `gpg` version on the CI/CD server **MUST MATCH** the one used locally. Otherwise, `gpg` decryption can fail silently, which leads to `git secret reveal` reporting `cannot find decrypted version of file` error. To be specific, `apt-get install gnupg` points to version [2.2.20](https://packages.ubuntu.com/impish/gnupg), yet `brew install gnupg` points to version [2.3.4](https://formulae.brew.sh/formula/gnupg) (as of 2022-01-17). Thus a `git-secret` encrypted file on macOS using the latest `gpg` installed from `brew` cannot be decrypted on Ubuntu (e.g. GitHub Actions' latest Ubuntu machine) using the latest `gpg` installed from `apt-get`. The work-around for this specific case is to downgrade `gpg` with `brew install gnupg@2.2.33`.
+
 
 ## Environment Variables and Configuration
 

